src/Stats.hs

```diff
@@ -97,12 +98,13 @@ data ReportResults =
         | DumpResult (TimeLog (CaptureData, TimeZone, ActivityData))
 
 
-filterPredicate :: [Filter] -> TimeLogEntry (Ctx, ActivityData) -> Bool
-filterPredicate filters tl =
-       all (\flag -> case flag of
-                Exclude act  -> excludeTag act tl
-                Only act     -> onlyTag act tl
-                GeneralCond s-> applyCond s (cTimeZone (fst (tlData tl))) M.empty tl) filters
+mkFilterPredicate :: [Filter] -> IO ApplyCond
+mkFilterPredicate filters = do
+    fs <- forM filters $ \flag -> case flag of
+                Exclude act  -> return $ excludeTag act
+                Only act     -> return $ onlyTag act
+                GeneralCond s-> mkApplyCond s
+    return $ \tl -> all ($ tl) fs
```

Comment:
I am a bit confused by this (but maybe I am just thick, haven’t looked at that code in a while). Previously, it was using the the timezone from `cTimeZone (fst (tlData tl))` – isn't that the timezone of the time log entry we are looking at?

Reply:
No, that is the timezone added by `prepare` in `readCategorize`, which is `getCurrentTimeZone`.

Reply:
But I am more confused by the code that was there before than by your code – in `readCategorizer` the `tz` that is passed to the parser is always `getCurrentTimeZone`, so presumably it should be the same when parsing the conditionals.

Reply:
So you are just too polite to say that the previous code (seemingly introduced in c81d5542e44f729cac74345f6096b2e754ac9a35) is kinda bogus? ;-)

Reply:
Yes, it is the same, since there is no other timezone anywhere else in the code.

But now that I'm looking some more into it, the Categorizer code seems to do the utc→local transformations during certain comparisons, so it seems like the support for possibly using the sample timezone is already partially there, it's just not being used since the sample timezone is not recorded and every Ctx just contains the local timezone. It's certainly something that can be added somewhat easily, but then your 10 years worth of logs wouldn't have this information anyway.

(I don't need this functionality right now because I rarely take myself nor my laptop outside of Europe, but if that ever changes, I can look into it. :-))